### ISSUE TITLE
Rest of file silently ignored after a certain string literal [PHP]

### ISSUE THREAD

**Issue by Krinkle** [doxygen/doxygen#9520]
**Describe the bug**
Rest of file silently ignored after a certain string literal.

**Expected behavior**

Warnings about a problem, or, for it to work as-is.

**Screenshots**

Code from https://github.com/wikimedia/mediawiki/blob/2ec4f75dd60ea687f23867d9170fa7e38aa75659/includes/MainConfigSchema.php output at https://doc.wikimedia.org/mediawiki-core/master/php/classMediaWiki_1_1MainConfigSchema.html. About half the file's members are missing from the documentation index.

I've attached an isolated example using a very short file, and a near-empty Doxyfile. In further brief, this is what the issue looks like:

```php
	/**
	 * This works.
	 */
	public const NamespaceAliases = [
		'default' => [],
		'type' => 'map',
	];

	/**
	 * This is and the rest of the file is ignored.
	 */
	public const LegalTitleChars = [
		'default' => ' %!"$&\'()*,\\-.\\/0-9:;=?@A-Z\\\\^_`a-z~\\x80-\\xFF+',
	];

	/**
	 * Gone.
	 */
	public const InterwikiExpiry = [
		'default' => 10800,
	];
```
<img width="1365" alt="Screenshot 2022-08-16 at 15 01 58" src="https://user-images.githubusercontent.com/156867/184899906-922183f7-6162-47d5-a68e-15868c4ca8f9.png">

**To Reproduce**
[Archive.zip](https://github.com/doxygen/doxygen/files/9351745/Archive.zip)

**Version**
1.9.1 is affected
1.9.5 is also affected.

**Additional context**

Looks related in impact to https://github.com/doxygen/doxygen/issues/8569. Might be unrelated in cause, though.

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #9521

**Comment by albert-github** [comment]
Small remark about the issue itself, when specifying a doxygen version and especially with the master version specify the full version from `doxygen -v`.
I saw the problem still present in 1.9.5 (d3a03e7a708e72a0c5a59da27e2be30d8f56c2a3 so based my work upon this version.

**Comment by albert-github** [comment]
Code has been integrated in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.5.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
